BUSCO updates

DLBPointon opened the pull request:
Addition of support for odb12 and a separate odb10 BUSCO module for ancestral busco mapping

DLBPointon committed:
Add seperate busco for ancestral thing, add odb12 support

DLBPointon commented:
That's the second container pull error in a row now, going to leave a bit of time and try again.

EDIT: Looks like it is actually a space issue

muffato commented:
> EDIT: Looks like it is actually a space issue

You can try using the larger runners like in one of the ToLA pipelines.

DLBPointon committed:
Update nf-test.yml

DLBPointon committed:
Update nf-test.yml

DLBPointon committed:
Update nf-test.yml

DLBPointon committed:
Update nf-test.yml

Change to 4c runner

DLBPointon commented:
> > EDIT: Looks like it is actually a space issue
> 
> You can try using the larger runners like in one of the ToLA pipelines.

Giving it a go, both this PR and the Pretext PR are now using `ubuntu2204-4c` although they havn't started yet.

muffato reviewed (commented):
That's a good start, and I've got not change to recommend on those files. However, something is missing.
By default, the pipeline finds the Busco lineage to use on its own. That's the `get_odb.py` script, which uses the lineage lists `assets/mapping_taxids-busco_dataset_name.eukaryota_odb10.2019-12-16.txt` (`--lineage_tax_ids` parameter, asset file is extracted from <https://busco-data.ezlab.org/v5/data/placement_files/mapping_taxids-busco_dataset_name.eukaryota_odb10.2019-12-16.txt.tar.gz>). The script  / config too need to be expanded for odb12.

DLBPointon committed:
Create mapping_taxids-busco_dataset_name.eukaryota_odb12.2025-01-15.txt

Addition of ODB12 mapping file

DLBPointon commented:
Ok, that makes sense.

I've just added the odb12 mapping file, I'll have a look at the other bits tomorrow.

DLBPointon commented:
@muffato 

For get_odb.py
I've added a get_odb_version function which simply returns _odb10 or _odb12 for this line:
`    odb_val = odb_arr[-1] + odb_version`

That seems like all it needs, is there anything else?

For the config
I've defaulted it to null and added the original values into the configs

Anything else?

DLBPointon committed:
update lineage tax id params

DLBPointon committed:
Add support for odb12

DLBPointon committed:
Forgot a :

muffato commented on nextflow_schema.json:
```diff
@@ -30,6 +30,11 @@
+                "ancestral_busco_lineage": {
+                    "type": "string",
+                    "description": "The odb10 used for the ancestral busco mapping (Only odb10 is currently supported).",
+                    "pattern": "._odb10$"
```
In what sense is only odb10 supported ? I wonder if the busco painter tool is actually compatible with both both odb10 and odb12, and the real dependency is on the format of the full table, and therefore the version of the Busco software rather than the version of the lineages used.

DLBPointon replied:
It's the ancestral mapping tsv file, it was made using odb10 busco assignments and then manually validated. My understanding was that going to odb12 changes the busco gene ids and so would invalidate the tsv? Obviously, if that's wrong then let me know.

Shane is going to talk to Mark about getting the reseachers to update them to odb12 to make life easier.

muffato replied:
Yes, the only TSV we have so far is based on lepidoptera_odb10, and it wouldn't work with lepidoptera_odb12. But equally, using the existing with TSV with mammalia_odb10 wouldn't work either. It's not really a matter of odb10 vs odb12. Both parameters (`--ancestral_table` and `--ancestral_busco_lineage`) have to be set consistently both in terms of lineage name and odb version, and our design decision in the original PR was to leave that to the users.
I think we wouldn't have to change anything in the pipeline if the TSV was for an odb12 lineage, and therefore would just remove any constraint of `--ancestral_busco_lineage` being an odb10.

DLBPointon replied:
Oh i see what you mean, what i've done it add a layer of control on something that doesn't really need it at the minute. In that case yeah I agree.

muffato reviewed (commented):
> That seems like all it needs, is there anything else?

That's an interesting take, I didn't think of that. So to switch between odbs, someone would have to change the path given to `--lineage_tax_ids` ?

> For the config I've defaulted it to null and added the original values into the configs

I don't see that change in the commits. `lineage_tax_ids` still has a value in `nextflow.config`, and the fact it's being set again on `conf/test*.config` is redundant.
But actually, I would like the pipeline to have a default value set so that we/users don't have to add `--lineage_tax_ids` every time.
The default value could be the odb12 file, I'm fine making the change now.

DLBPointon commented:
> That's an interesting take, I didn't think of that. So to switch between odbs, someone would have to change the path given to `--lineage_tax_ids` ?
> 

I thought it was the simplest solution. Stops the need for another flag to set the odb type and such.

> I don't see that change in the commits. `lineage_tax_ids` still has a value in `nextflow.config`, and the fact it's being set again on `conf/test*.config` is redundant. But actually, I would like the pipeline to have a default value set so that we/users don't have to add `--lineage_tax_ids` every time. The default value could be the odb12 file, I'm fine making the change now.

I definitely added it, but yeah, it's gone on my side too. If you're free, then go ahead. Will it make sense to default a `ancestral_busco_lineage = "lepidoptera_odb10"` or is that unnecessary for now.

Just need to update the snap after that's updated, and done.

muffato committed:
Switching to odb12 for the main assessment

muffato committed:
Revert "update lineage tax id params"

This reverts commit 52d70f399494c68e4eb4b43956308a1b81470a51.

muffato committed:
nf-test udpate

muffato committed:
Constraint is not needed

muffato committed:
Shifted the line numbers to match the expected output

muffato committed:
nf-test update
